lib/request.ts

```diff
@@ -94,10 +98,17 @@ export const request = (
     }
   }
 
+  // connection timeout - Happens when the socket connection cannot be established
+  const connectionTimeout = options.timeout || CONNECTION_TIMEOUT;
+  /**
+   * Read timeout - Happens after the socket connection is successfully established and
+   * when there is no activity on the connected socket.
```

Comment:
from the description, this sounds more like `idleTimeout` to me?

Reply:
both can mean idletimeout, when we connect we wait, when we try to read we wait.